### PR TITLE
Add bool support for hdf5

### LINKED ISSUE #1144: Loading boolean data from hdf5
Hi I have the following error when loading HDF5 boolean data with the latest versions: 

```
InvalidArgumentError: unsupported data class for /my_bool_data: 8 [Op:IO>HDF5ReadableInfo]
```

Here is a [link](https://colab.research.google.com/drive/11sUZgDcfQLRgG7PeE2PXBJIShR_1i6qH?usp=sharing) to a google colab.

### PR DESCRIPTION
**Pull request by yongtang**
This PR adds bool support for hdf5.

This PR fixes #1144.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Review by terrytangyuan — APPROVED** [review]

File: tensorflow_io/core/kernels/hdf5_kernels.cc [hunk 1]
```diff
@@ -243,6 +243,53 @@ class HDF5ReadableResource : public ResourceBase {
...
+                  static_cast<H5::EnumType&>(data_type).nameOf(&value, 100);
+              if (ii != 0) {
+                names += ", ";
+              }
+              names += name;
+            }
+            names += "]";
+            return errors::InvalidArgument("unsupported data class for enum :",
+                                           names);
+          }
```
Comment (kvignesh1420) [inline tensorflow_io/core/kernels/hdf5_kernels.cc:289]: @yongtang for which kind of data will this block be executed?

File: tensorflow_io/core/kernels/hdf5_kernels.cc [hunk 2]
```diff
@@ -455,6 +502,65 @@ class HDF5ReadableResource : public ResourceBase {
...
+                  if (ii != 0) {
+                    names += ", ";
+                  }
+                  names += name;
+                }
+                names += "]";
+                return errors::InvalidArgument(
+                    "unsupported data class for enum :", names);
+              }
+            }
```
Comment (kvignesh1420) [inline tensorflow_io/core/kernels/hdf5_kernels.cc:554]: same query as above. A test case which can raise this error would be great.

**Comment by yongtang** [comment]
@kvignesh1420 On hdf5, there is no dedicated `BOOL` data types. Instead, by convention hdf5 uses a `ENUM` type of `["FALSE"=0, "TRUE"=1]` to represent `BOOL` data types. In this PR only `BOOL` type is handled (not the generic `ENUM` type). I have updated the PR to add additional test where a generic `ENUM` type will results in an exception. Please take a look.

**Review by kvignesh1420 — APPROVED** [review]
LGTM.

**Pull request merged**